streaming/storage/download.py

```diff
@@ -474,18 +475,25 @@ def download_file(remote: Optional[str], local: str, timeout: float):
         download_from_local(remote, local)
 
 
-def wait_for_download(local: str, timeout: float = 60) -> None:
-    """Wait for a download by another thread/process to complete.
+def wait_for_file_to_exist(filename: str, poll_interval: float, timeout: float,
+                           err_msg: str) -> None:
+    """Wait for the file to exist till timeout seconds. Raise an Exception after that.
 
     Args:
-        local (str): Local path (local filesystem).
-        timeout (float): How long to wait for file to download before raising an exception.
-            Defaults to ``60``.
+        filename (str): A file name
+        poll_interval (float): Number of seconds to wait before next polling
+        timeout (float): Number of seconds to wait for a file to exist before raising an exception
+        err_msg (str): Error message description for an exception
+
+    Raises:
+        RuntimeError: Raise an Exception if file does not exist after timeout
     """
-    t0 = time()
-    while not os.path.exists(local):
-        elapsed = time() - t0
-        if timeout < elapsed:
-            raise TimeoutError(
-                f'Waited longer than {timeout}s for other worker to download {local}.')
-        sleep(0.25)
+    start_time = time()
+    while True:
+        sleep(poll_interval)
+        if os.path.exists(filename):
+            sleep(poll_interval)
```

Comment:
Do we need to sleep here since the file already exists now?

Reply:
Yeah that is mysterious... from streaming experience, I wonder if this line is for a race condition where a small file is getting written by someone else. Would prefer to be conservative for the time being...